### PR TITLE
data/d10_1038_nature22330

### PR DESCRIPTION
**Pull request by irena-starters**

### CODE REVIEW

File: sfaira/data/dataloaders/loaders/d10_1038_nature22330/homosapiens_forebrain_2017_microwellseq_birey_001.yaml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    assay_sc: "microwell-seq"
+    assay_sc_obs_key: 
+    assay_differentiation: "Pasca, 2015" 
+    assay_differentiation_obs_key: 
+    assay_type_differentiation: "guided"
+    assay_type_differentiation_obs_key: 
+    bio_sample:
+    bio_sample_obs_key: 
+    cell_line: 
+    cell_line_obs_key:
```
Comment (le-ander) [inline sfaira/data/dataloaders/loaders/d10_1038_nature22330/homosapiens_forebrain_2017_microwellseq_birey_001.yaml:36]: I would say the cell line information is quite crucial here. it seems really hard to identify from the paper though. can you try to dig a little to see if we can somehow define the cell_line that was used for these organoids? (supplementary table 1 states that both the lines 2242-1 and 8858-1 were used for the single-cell. but it's not quite clear to me where each of these was used.) if you cannot find the information you can maybe email the corresponding author and ask them.

File: sfaira/data/dataloaders/loaders/d10_1038_nature22330/homosapiens_forebrain_2017_microwellseq_birey_001.yaml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    assay_differentiation_obs_key: 
+    assay_type_differentiation: "guided"
+    assay_type_differentiation_obs_key: 
+    bio_sample:
+    bio_sample_obs_key: 
+    cell_line: 
+    cell_line_obs_key: 
+    cell_type: 
+    cell_type_obs_key: 
+    development_stage: "14th week post-fertilization human stage"
```
Comment (le-ander) [inline sfaira/data/dataloaders/loaders/d10_1038_nature22330/homosapiens_forebrain_2017_microwellseq_birey_001.yaml:39]: wouldn't day 105 be 15th week?
105/7=15

File: sfaira/data/dataloaders/loaders/d10_1038_nature22330/homosapiens_forebrain_2017_microwellseq_birey_001.yaml [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    development_stage_obs_key: 
+    disease: 
+    disease_obs_key: 
+    ethnicity: 
+    ethnicity_obs_key: 
+    gm:
+    gm_obs_key:
+    individual:
+    individual_obs_key: 
+    organ: "forebrain"
```
Comment (le-ander) [inline sfaira/data/dataloaders/loaders/d10_1038_nature22330/homosapiens_forebrain_2017_microwellseq_birey_001.yaml:49]: if you specify "organ_obs_key", then "organ" must be empty. you either specify these terms per cell (ie. via obs key) or per dataset

File: sfaira/data/dataloaders/loaders/d10_1038_nature22330/homosapiens_forebrain_2017_microwellseq_birey_001.yaml [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    organism_obs_key: 
+    sample_source: "3d_culture"
+    sample_source_obs_key: 
+    sex: "male"
+    sex_obs_key: 
+    source_doi:
+    source_doi_obs_key:
+    state_exact:
+    state_exact_obs_key:
+    suspension_type:
```
Comment (le-ander) [inline sfaira/data/dataloaders/loaders/d10_1038_nature22330/homosapiens_forebrain_2017_microwellseq_birey_001.yaml:61]: You can add "cell" here as the protocol is single cell and not single nucleus

File: sfaira/data/dataloaders/loaders/d10_1038_nature22330/homosapiens_forebrain_2017_microwellseq_birey_001.yaml.save [hunk 5]
```diff
@@ -0,0 +1,91 @@
+dataset_structure:
```
[the remark this answers is not in the document]
Comment (irena-starters) [reply sfaira/data/dataloaders/loaders/d10_1038_nature22330/homosapiens_forebrain_2017_microwellseq_birey_001.yaml.save:1]: Removed it

File: sfaira/data/dataloaders/loaders/d10_1038_nature22330/homosapiens_forebrain_2017_microwellseq_birey_001.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import numpy as np
+import pandas as pd
+import tarfile
+import anndata as ad
+import scipy.sparse
+import os
+# import any packages you require for dataloading here. you can assume packages like numpy and pandas being available
```
[the remark this answers is not in the document]
Comment (irena-starters) [reply sfaira/data/dataloaders/loaders/d10_1038_nature22330/homosapiens_forebrain_2017_microwellseq_birey_001.py:7]: Removed it.

File: sfaira/data/dataloaders/loaders/d10_1038_nature22330/homosapiens_forebrain_2017_microwellseq_birey_001.yaml [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+    layer_velocity: 
+dataset_or_feature_wise:
+    feature_reference: 
+    feature_reference_var_key: 
+    feature_type: "rna"
+    feature_type_var_key: 
+dataset_or_observation_wise:
+    assay_sc: "microwell-seq"
+    assay_sc_obs_key: 
+    assay_differentiation: "Pasca, 2015"
```
[the remark this answers is not in the document]
Comment (irena-starters) [reply sfaira/data/dataloaders/loaders/d10_1038_nature22330/homosapiens_forebrain_2017_microwellseq_birey_001.yaml:29]: Added.

File: sfaira/data/dataloaders/loaders/d10_1038_nature22330/homosapiens_forebrain_2017_microwellseq_birey_001.yaml [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+    feature_type_var_key: 
+dataset_or_observation_wise:
+    assay_sc: "microwell-seq"
+    assay_sc_obs_key: 
+    assay_differentiation: "Pasca, 2015" 
+    assay_differentiation_obs_key: 
+    assay_type_differentiation: "guided"
+    assay_type_differentiation_obs_key: 
+    bio_sample:
+    bio_sample_obs_key:
```
[the remark this answers is not in the document]
Comment (irena-starters) [reply sfaira/data/dataloaders/loaders/d10_1038_nature22330/homosapiens_forebrain_2017_microwellseq_birey_001.yaml:34]: Makes sense.

File: sfaira/data/dataloaders/loaders/d10_1038_nature22330/homosapiens_forebrain_2017_microwellseq_birey_001.yaml [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+layers:
+    layer_counts: "X"
+    layer_processed: 
+    layer_spliced_counts: 
+    layer_spliced_processed: 
+    layer_unspliced_counts: 
+    layer_unspliced_processed: 
+    layer_velocity: 
+dataset_or_feature_wise:
+    feature_reference:
```
[the remark this answers is not in the document]
Comment (irena-starters) [reply sfaira/data/dataloaders/loaders/d10_1038_nature22330/homosapiens_forebrain_2017_microwellseq_birey_001.yaml:22]: I understood that it had to be the exact reference used in the paper, and not the latest one.

File: sfaira/data/dataloaders/loaders/d10_1038_nature22330/homosapiens_forebrain_2017_microwellseq_birey_001.yaml (hunk 2 above)
Comment (irena-starters) [reply sfaira/data/dataloaders/loaders/d10_1038_nature22330/homosapiens_forebrain_2017_microwellseq_birey_001.yaml:39]: True, I have changed it.

File: sfaira/data/dataloaders/loaders/d10_1038_nature22330/homosapiens_forebrain_2017_microwellseq_birey_001.yaml (hunk 9 above)
[the remark this answers is not in the document]
Comment (le-ander) [reply sfaira/data/dataloaders/loaders/d10_1038_nature22330/homosapiens_forebrain_2017_microwellseq_birey_001.yaml:22]: ah, I found it. seems like the gencode hg19 release maps to ensembl 74: https://www.gencodegenes.org/human/releases.html

So it should be: `Homo_sapiens.GRCh37.75`

File: sfaira/data/dataloaders/loaders/d10_1038_nature22330/homosapiens_forebrain_2017_microwellseq_birey_001.yaml (hunk 3 above)
Comment (le-ander) [reply sfaira/data/dataloaders/loaders/d10_1038_nature22330/homosapiens_forebrain_2017_microwellseq_birey_001.yaml:49]: "sample" should also be listed as "organ_obs_key" as the different sample correspond to different organs.
just remoce the forebrain label for the whole dataset

Comment (irena-starters) [reply sfaira/data/dataloaders/loaders/d10_1038_nature22330/homosapiens_forebrain_2017_microwellseq_birey_001.yaml:49]: Alright.

File: sfaira/data/dataloaders/loaders/d10_1038_nature22330/homosapiens_forebrain_2017_microwellseq_birey_001.py [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+    # replace the simple data loading code below with the code required to load your data file(s)
+    dfs = []
+    obs = []
+    with tarfile.open(fn) as tar:
+        for member in tar.getmembers():
+            d = pd.read_csv(tar.extractfile(member), compression="gzip", header=0, index_col=None)
+            for i in range(d.shape[0]):
+                obs.append(member.name.split("_")[1])
+            dfs.append(d)
+    adata = ad.AnnData(X=scipy.sparse.csr_matrix(np.vstack((dfs[0], dfs[1]))), obs=pd.DataFrame({"sample": obs}), var=pd.DataFrame(index=dfs[0].columns.values))
```
Comment (le-ander) [inline sfaira/data/dataloaders/loaders/d10_1038_nature22330/homosapiens_forebrain_2017_microwellseq_birey_001.py:23]: One more thing: could you add a line here that adds a column to obs with the organoid age in days inside see my message on mattermost).
```python
adata.obs["organoids_age_days"] = "105"
```

Comment (irena-starters) [reply sfaira/data/dataloaders/loaders/d10_1038_nature22330/homosapiens_forebrain_2017_microwellseq_birey_001.py:19]: Done.

**Pull request closed without merging**